CHANGELOG.md

```diff
@@ -1,5 +1,125 @@
 # Changelog
 
+## unreleased
+
+[Full Changelog](https://github.com/delta-io/delta-kernel-rs/compare/v0.4.0...HEAD)
+
+**API Changes**
+
+*Breaking*
+
+1. `Expression::Column(String)` is now `Expression::Column(ColumnName)` [\#400]
+2. delta_kernel_ffi::expressions moved into two modules:
+   `delta_kernel_ffi::expressions::engine` and `delta_kernel_ffi::expressions::kernel` [\#363]
+3. FFI: removed (hazardous) `impl From` for `KernelStringSlize` and added `unsafe` constructor
+   instead [\#441]
+4. Moved `LogSegment` into its own module (`log_segment::LogSegment`) [\#438]
+5. `DvInfo` now implements `Clone`, `PartialEq`, and `Eq` [\#468]
+6. `Stats` now implements `Debug`, `Clone`, `PartialEq`, and `Eq` [\#468]
+7. Renamed `EngineData::length` as `EngineData::len` [\#471]
+8. New `AsAny` trait: `AsAny: Any + Send + Sync` required bound on all engine traits [\#450]
+9. Rename `mod features` to `mod table_features` [\#454]
+10. LogSegment fields renamed: `commit_files` -> `ascending_commit_files` and `checkpoint_files` ->
+    `checkpoint_parts` [\#495]
+11. Added minimum-supported rust version: currenly rust 1.80 [\#504]
+12. Improved row visitor API: renamed `EngineData::extract` as `EngineData::visit_rows`, and
+    `DataVisitor` trait renamed as `RowVisitor` [\#481]
+13. FFI: New `mod engine_data` and `mod error` (moved `Error` to `error::Error`) [\#537]
+14. new error types (TODO)
+
+*Additions*
+
+1. New `ColumnName`, `column_name!`, `column_expr!` for structured column name parsing. [\#400]
+   [\#467]
+2. New `Engine` API `write_json_file()` for atomically writing JSON [\#370]
+3. New `Transaction` API for creating transactions, adding commit info and write metadata, and
+   commiting the transaction to the table. Includes `Table.new_transaction()`,
+   `Transaction.write_context()`, `Transaction.with_commit_info`, `Transaction.with_operation()`,
+   `Transaction.with_write_metadata()`, and `Transaction.commit()` [\#370] [\#393]
+4. FFI: Visitor for converting kernel expressions to engine expressions. See the new example at
+   `ffi/examples/visit-expression/` [\#363]
+5. FFI: New `TryFromStringSlice` trait and `kernel_string_slice` macro [\#441]
+6. New `DefaultEngine` engine implementation for writing parquet: `write_parquet_file()` [\#393]
+7. Added support for parsing comma-separated column name lists:
+   `ColumnName::parse_column_name_list()` [\#458]
+9. New `VacuumProtocolCheck` table feature [\#454]
+10. Added `Cdc` action support [\#506]
+11. (early CDF read support) New `TableChanges` type to read CDF from a table between versions
+    [\#505]
+12. (early CDF read support) Builder for scans on `TableChanges` [\#521]
+13. New `TableProperties` struct which can parse tables' `metadata.configuration` [\#453] [\#536]
+
+**Implemented enhancements:**
+- FFI examples now use AddressSanitizer [\#447]
```

Comment:
Nit: How come we use `-` for these bullet points, but numbers in the Additions?

Reply:
no reason lol other than it felt nice to have structure in the changes/additions and felt less useful here. want numbers everwhere?

Reply:
I have a slight preference towards a consistent bullets, but I'm cool with either.